core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -290,6 +290,18 @@ oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
     Topic.create = function(topicBackendObject) {
       return new Topic(topicBackendObject);
     };
+
+    // Create a new, empty topic. This is not guaranteed to pass validation
+    // tests.
+    Topic.createEmptyTopic = function() {
```

Comment:
When I was doing the topiceditor, I saw that in collections, when the StateService is initialised, an empty object was initialised, after which the collection is set, when _setCollection() is called, the collection is actually set, so this would be needed.

Reply:
Ah OK. In that case, leave a comment about this here please. Perhaps:

```
// Create a new, empty topic that is not guaranteed to pass validation tests. This should only be used
// when initializing the topic editor (prior to setting it with data from the backend).
```

That said ... this seems a little weird to me. Why do we have this intermediate phase where the object is invalid? It doesn't seem to me that a blank topic is much better than a null value. I'll send an email to @BenHenning to check, but it's fine to keep this in parity with the collection editor for now.

Reply:
Ok.

Reply:
Followed up via email, but to summarize on here also for posterity:

It's been a while since that code was changed, but I think the idea is that a null collection needs to be special cased, but an empty collection can still be valid. It's not clear to me if that invariant actually holds here, but given that we don't check anywhere if collection == empty collection (compared to checking if it's null), I suspect it does. This seems safer in general since it means that the editor needs to be properly support empty collections, and that in turn offers more predictability: the editor goes from showing a default empty collection to a properly loaded collection.

If the empty collection doesn't pass validation tests, that may be an oversight. Empty objects should be valid insomuch as they can be displayed properly, but it may not be valid to save them (that's okay--starting off with an empty collection is valid even if it can't be saved since it provides an initial space to start working).

To put another way, whatever default value is used in editors should satisfy:
- Facilitating some sort of default view without requiring special casing and without errors
- Not being saveable without creator interaction (this holds true for explorations and possibly collections--not as sure on the latter)

We could probably do a better job here, but I think that's the rough rational behind not using null.